File: backend/store/app/order/query/book.go
package query

import "store/app/domain/data"

type Book struct {
	Id           string
	Title        string
	Subtitle     string
	Description  string
	ThumbnailUrl string
	OnhandQty    int
	ReservedQty  int
}

func (Book) fromDataObject(b *data.Book) *Book {
	return &Book{
		Id:           b.Id,
		Title:        b.Title,
		Subtitle:     b.Subtitle,
		Description:  b.Description,
		ThumbnailUrl: b.ThumbnailUrl,
		ReservedQty:  b.ReservedQty,
		OnhandQty:    b.OnhandQty,
	}
}

File: backend/ecommerce/src/main/java/com/bookstore/ecommerce/app/order/query/model/Book.java
package com.bookstore.ecommerce.app.order.query.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

@AllArgsConstructor
@Builder
@Data
public class Book {
  private String id;
  private String title;
  private String subtitle;
  private String description;
  private String[] authors;
  private double averageRating;
  private int ratingsCount;
  private String thumbnailUrl;
  private String previewUrl;
  private int onhandQty;
  private int reservedQty;

  public static Book fromDataObject(com.bookstore.ecommerce.app.domain.data.Book book) {
    return Book.builder()
      .id(book.getId())
      .title(book.getTitle())
      .subtitle(book.getSubtitle())
      .description(book.getDescription())
      .build();
  }
}
